app/src/main/java/com/futurice/freesound/network/api/ApiNetworkModule.java

```diff
@@ -69,20 +65,12 @@ static FreeSoundApi provideFreeSoundApi(@Named(API_URL_CONFIG) String url,
     @Provides
     @Singleton
     @ForFreeSoundApi
-    static Gson provideGson() {
-        return new GsonBuilder()
-                .setDateFormat(ApiConstants.DATE_FORMAT_PATTERN)
-                .registerTypeAdapterFactory(FreesoundTypeAdapterFactory.create())
-                .create();
-    }
-
-    @Provides
-    @Singleton
-    @ForFreeSoundApi
     static Moshi provideMoshi() {
         return new Moshi.Builder()
                 .add(new KotlinJsonAdapterFactory())
                 .add(new GeoLocationJsonAdapter())
+                .add(new DateAdapter())
```

Comment:
When using only `Rfc3339DataJsonAdapter`, the parser correctly throws on this code:

```java
   // extract timezone
      if (date.length() <= offset) {
        throw new IllegalArgumentException("No time zone indicator");
      }
```

and then throws from here:

```java
      throw new JsonDataException("Not an RFC 3339 date: " + date);
```

Weird that it's failing silently afterwards: no stacktrace, no log, no error indicator. The error is definitely propagated out of Retrofit though.